Expire connection after 1800s ~ 5400s

mwtian opened the pull request:
This change shows to be effective at avoiding memory usage growth, capping node memory usage below 10GB.
<img width="1820" alt="image" src="https://user-images.githubusercontent.com/81660174/203108507-42e78a00-cd27-44d7-a767-19b7ec3bcd6b.png">

The above experiment ran for ~8 hours in private testnet, with the typical stress client load of 50/s shared and 50/s owned transactions. There is essentially no memory usage growth of the `sui-node` process when observed from hosts. The process vsize is ~3.5GB and rss is ~1.8GB after 8hr. Before, the process vsize can grow at the rate of ~3GB/hr, and after running for ~8hr the process vsize can be >20GB.

The experiment also used a more aggressive timeout at 1min across all nodes without random jitter. Under this more aggress timeout, Narwhal still seemed to work well. Although Narwhal should maintain liveness and other properties with occasionally broken connections or failed requests, a longer expiration time range of 30min ~ 90min is chosen to avoid all connections broken at the same time disrupting the network.

akichidis commented:
ok this is super interesting graph @mwtian and promising PR! So from your PR I am thinking that the lines we are looking at:
* somehow there is a memory issue related to a long running quinn connection it self
* managing messages in `InboundRequestHandler` it self leading to memory leak (also related to what @gdanezis has figured [out here](https://github.com/bmwill/anemo/pull/3) ? )

aside from that would be curious to see also any metrics/data around the size of `inflight_requests` JoinSet

also for how long does your PR run and what's the load on the testing environment? What's the comparison to the memory consumption before the PR?

mwtian commented:
It is a bit hard to plumb metrics through anemo, but I have added logs in an experiment and `inflight_requests` JoinSet / FuturesUnordered has expected size (<100) and regularly goes to empty.

Added the experiment info to the PR description, which I will copy below:

> The above experiment ran for ~8 hours in private testnet, with the typical stress client load of 50/s shared and 50/s owned transactions. There is essentially no memory usage growth of the `sui-node` process when observed from hosts. The process vsize is ~3.5GB and rss is ~1.8GB after 8hr. Before, the process vsize can grow at the rate of ~3GB/hr, and after running for ~8hr the process vsize can be >20GB.

It will definitely be interesting to investigate how Connection seems to retain memory.
